client/src/app/components/audit-template/audit-template-view/create-audit-template/create-audit-template.component.ts

```diff
@@ -30,26 +29,83 @@ export class CreateAuditTemplateComponent extends AuditTemplateViewComponent {
       bins: '',
       part_number: '',
       serial_number: '',
+      start_date: '',
+      repeat_every: this.repeatEvery,
+      on_day: this.dayArray,
+      for_month: this.monthArray,
+      time_zone_utc: this.timeZoneUTC,
     };
   }
 
   submitQuery(body: Template): void {
-    this.auditTemplateService.createTemplate(body).subscribe(
-      () => {
-        setTimeout(() => {
-          // Redirect user back to list of templates
-          this.router.navigate(['template']);
-        }, 1000); // Waiting 1 second before redirecting the user
-        this.initializeForm();
-        this.errorMessage = '';
-
-      },
-      (err) => {
-        // if backend returns an error
-        if (err.error) {
-          this.errorMessage = err.error;
+    this.dayArray = [];
+    this.monthArray = [];
+    const year = this.startDate.getFullYear();
+    const month = ((this.startDate.getMonth() + 1).toString().length === 1) ?
+      ('0' + (this.startDate.getMonth() + 1)) : ((this.startDate.getMonth() + 1));
+    const day = (this.startDate.getDate().toString().length === 1) ? ('0' + (this.startDate.getDate())) : (this.startDate.getDate());
+    const hour = (parseInt(this.startTime.split(':')[0], 10).toString().length === 1) ?
+      ('0' + (parseInt(this.startTime.split(':')[0], 10))) : (parseInt(this.startTime.split(':')[0], 10));
+    const minute = (parseInt(this.startTime.split(':')[1], 10).toString().length === 1) ?
+      ('0' + (parseInt(this.startTime.split(':')[1], 10))) : (parseInt(this.startTime.split(':')[1], 10));
```

Comment:
This gives me a headache to look at and try to understand. Can we take each of these statements and inline them with a function instead? This would make them much more readable if the function names are properly defined.

Reply:
This has been fixed

Reply:
I see what you did, but I would prefer each of these assignments were done with their own function instead of dumping the functionality into an already long function.

Reply:
Oops, I see what you meant! Sorry I misunderstood earlier. This has been fixed now 😄